test/hiccup2/optimizations_test.clj

```diff
@@ -0,0 +1,90 @@
+(ns hiccup2.optimizations-test
+  (:require [clojure.pprint :as pp]
+            [clojure.test :refer :all]
+            [clojure.walk :as walk]
+            [hiccup2.core :as h]))
+
+(defn- count-forms [data]
+  (let [*counter (atom 0)]
+    (walk/postwalk (fn [x]
+                     (when (seq? x)
+                       (swap! *counter inc))
+                     x)
+                   data)
+    @*counter))
+
+(deftest test-count-forms
+  (is (= 1 (count-forms '())))
+  (is (= 3 (count-forms '(if (number? x) (inc x) x)))))
```

Comment:
I don't think we need to regression test our regression tests, so these can be omitted.